kubel.el

```diff
@@ -903,15 +908,9 @@ P can be a single number or a localhost:container port pair."
                       (when (tramp-tramp-file-p default-directory)
                         (with-parsed-tramp-file-name default-directory nil
                           (format "%s%s:%s@%s|" (or hop "") method user host))) ""))
-         (pod (if (kubel--is-pod-view)
-                  (kubel--get-resource-under-cursor)
-                (kubel--select-resource "Pods")))
-         (containers (kubel--get-containers pod))
-         (container (if (equal (length containers) 1)
-                        (car containers)
-                      (completing-read "Select container: " containers)))
-         (default-directory (format "/%skubectl:%s@%s:/" dir-prefix container pod)))
-    (shell (format "*kubel - shell - %s@%s*" container pod))))
+         (con-pod (kubel--get-container-under-cursor))
+         (default-directory (format "/%skubectl:%s@%s:/" dir-prefix (car con-pod) (cdr con-pod))))
```

Comment:
nitpick: since you are inside a `let*`, you could do a 
```
(container (car con-pod))
(pod (cdr con-pod))
```
in there to reduce some of the car/cdr later and make things more readable.

Reply:
Thanks, I refactored